src/mbgl/text/cross_tile_symbol_index.cpp

```diff
@@ -112,8 +131,28 @@ bool CrossTileSymbolLayerIndex::addBucket(const OverscaledTileID& tileID,
         }
     }
 
-    for (auto& symbolInstance: bucket.symbolInstances) {
-        symbolInstance.crossTileID = 0;
+    bucket.hasUninitializedSymbols = false;
+
+    if (tileID.overscaleFactor() > 1u) {
+        // For overscaled tiles the viewport might be showing only a small part of the tile,
+        // so we filter out the off-screen symbols to improve the performance.
+        mat4 posMatrix;
+        const auto& projMatrix = transformState.getProjectionMatrix();
```

Comment:
Reconsider reusing [renderTile.matrix calculation](https://github.com/mapbox/mapbox-gl-native/blob/c7bbd5bbef0780df5d0a5498e14051ee63c24e44/src/mbgl/renderer/render_tile.cpp#L124) and passing it here from `CrossTileSymbolIndex::addLayer`, also in https://github.com/mapbox/mapbox-gl-native/blob/9bb1f898030a8bf73c11e53f498684e24a61ce60/src/mbgl/text/placement.cpp#L157 as all three might calculate the same.

Reply:
Great point, thanks!